jobsdb/jobsdb.go

```diff
@@ -1412,11 +1414,16 @@ func computeInsertVals(before, after []string) ([]string, error) {
 				return calculatedVals, fmt.Errorf("computed index is invalid. before: %v, after: %v, calculatedVals: %v", before, after, calculatedVals)
 			}
 		}
-
+		var comparatorBool bool
+		if skipZeroAssertionForMultitenant {
+			comparatorBool = len(calculatedVals) == 2
+		} else {
+			comparatorBool = (before[0] == "0" && len(calculatedVals) == 3) ||
+				(before[0] != "0" && len(calculatedVals) == 2)
+		}
 		//Only when the index starts with 0, we allow three levels. This would be when we have to insert an internal migration DS between two import DSs
```

Comment:
this comment doesn't seem to apply to the `skipZeroAssertion` case

Reply:
modified the comment